service/actions/websocket.js

```diff
@@ -224,7 +224,7 @@ async function handleNewPostEvent(msg, dispatch, getState) {
         }
     ]), getState);
 
-    if (userId === users.currentId || post.channel_id === currentChannelId || isSystemMessage(post)) {
+    if ((userId === users.currentId && !isSystemMessage(post)) || post.channel_id === currentChannelId || shouldIgnorePost(post)) {
```

Comment:
There's a not insignificant problem here:
If a user receives an ignored message (e.g. `system_add_to_channel`) in a channel that is unread it will mark the channel as read.